msgraph-developer-proxy-plugins/MockResponses/MockResponsePlugin.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Graph.DeveloperProxy.Abstractions;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Net;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using Titanium.Web.Proxy.EventArguments;
+using Titanium.Web.Proxy.Http;
+using Titanium.Web.Proxy.Models;
+
+namespace Microsoft.Graph.DeveloperProxy.Plugins.MocksResponses;
+
+internal class MockResponseConfiguration {
+    public bool NoMocks { get; set; } = false;
+    public string MocksFile { get; set; } = "responses.json";
+
+    [JsonPropertyName("responses")]
+    public IEnumerable<MockResponse> Responses { get; set; } = Array.Empty<MockResponse>();
+}
+
+public class MockResponsePlugin : IProxyPlugin {
+    private ISet<Regex>? _urlsToWatch;
+    private ILogger? _logger;
+    private MockResponseConfiguration _configuration = new();
+    private MockResponsesLoader? _loader = null;
+    private readonly Option<bool> _noMocks;
+    public string Name => nameof(MockResponsePlugin);
+
+    public MockResponsePlugin() {
+        _noMocks = new Option<bool>("--no-mocks", "Disable loading mock requests");
+        _noMocks.AddAlias("-n");
+        _noMocks.ArgumentHelpName = "no mocks";
+        _noMocks.SetDefaultValue(false);
```

Comment:
There should be a command line option to allow developers to pass in the name/path of the mocks file at runtime.

Reply:
Right now, we're using a fixed name. Shall we add a new issue to make it configurable?

Reply:
I already made it configurable in this changeset: https://github.com/microsoftgraph/msgraph-developer-proxy/pull/99/files#diff-d162269ed5127e480a21c801717a78de8a328e3ffc781d0514ba50bcb8c3df04R18